internal/sqlsmith: TestGenerateParse failed

internal/sqlsmith.TestGenerateParse [failed](https://mesolite.cluster.engflow.com/invocations/default/c0648aae-7b21-4a1a-bf23-2656875bef70?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvaW50ZXJuYWwvc3Fsc21pdGg6c3Fsc21pdGhfdGVzdA==) on master @ [3e855cb8fa1ec36916bcd867d666499322cb9ab7](https://github.com/cockroachdb/cockroach/commits/3e855cb8fa1ec36916bcd867d666499322cb9ab7):


```
					AS tab_70 ("co/l""_153")
		)
SELECT
	NULL AS col_154,
	" tab_71"._jsonb AS "c😯o{l_155",
	" tab_71"._uuid AS "c%vol_156",
	e'\\x7a68':::BYTES AS col_157,
	" tab_71".tableoid AS "c%vol_158",
	" tab_71".crdb_internal_origin_id AS col_159,
	3071379683945844277:::INT8 AS "c ol_160",
	(-29078):::INT8 AS "�c %vol_161"
FROM
	defaultdb.public.seed AS " tab_71", with_9 AS cte_ref_3
ORDER BY
	" tab_71"._float8 NULLS FIRST, cte_ref_3."co/l""_153", " tab_71".tableoid DESC NULLS FIRST
LIMIT
	20:::INT8;

STMT: 39
DELETE FROM
	defaultdb.public.seed AS tab_72
USING
	defaultdb.public.seed AS "t'ab_73", defaultdb.public.seed AS tab_74
LIMIT
	53:::INT8;

STMT: 40
BACKUP TABLE defaultdb.public.seed INTO 'http://127.0.0.1:39397/backup_2' WITH revision_history = true;

STMT: 41
DELETE FROM
	defaultdb.public.seed AS tab😐_75
USING
	defaultdb.public.seed AS tab_76, defaultdb.public.seed AS tab_77
LIMIT
	54:::INT8;

STMT: 42
DELETE FROM defaultdb.public.seed AS tab_78 WHERE NULL LIMIT 99:::INT8;

STMT: 43
UPDATE defaultdb.public.seed AS tab_79 SET _int8 = 7067579187763201781:::INT8 WHERE tab_79._string ~* tab_79._string;

STMT: 44
BACKUP TABLE defaultdb.public.seed INTO e'http://127.0.0.1:39397/backup\t_3' WITH revision_history = false;

    setup_test.go:116: RESTORE TABLE defaultdb.public.seed FROM 'http://127.0.0.1:39397/Backup_1' WITH into_db = 'into_db': at or near "with": syntax error: The ``RESTORE <targets> FROM <backupURI>`` syntax is no longer supported. Please use ``RESTORE <targets> FROM <subdirectory> IN <collectionURI>``.
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestGenerateParse2463339066
--- FAIL: TestGenerateParse (1.79s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-43990

internal/sqlsmith.TestGenerateParse [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17594848?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17594848?buildTab=artifacts#/) on master @ [65ecc021945ea78acedd7fe264f3d65245e3ac2a](https://github.com/cockroachdb/cockroach/commits/65ecc021945ea78acedd7fe264f3d65245e3ac2a):


```
							'[399679.1,1.7419285e+16,2.5180478e+17,4.4616057e+34,-7.755248e+36,-0.024939284,3.566963e+16,-0.005968388,0.014049233,3.6581938e-25,2.064296e-12,1.2215983e-26,-2.5430364e-22,-5.925874e+35,-1.3842526e-07,4.582354e-21,1.695666,-2.8338107e-07,-13.559997,-1.3810473e+34,2.5932105e-30,-1.6775976e+12,2.442852e-30,0.0026778353,-1.3797843e-14,-7.7880156e+30,-7.193662e-30,-4.9605944e-19,6.055113e-07,-4.8882458e-15,-3.4995933e-24,2.0440258e+21,-5.5820017e+23,7.7668235e-29,9.2188466e+33,-3.9151222e+35,-4.3554323e+26,-1.449819e+16,-2.3720353,-1.7096646e+34,-104.34633,-4.2267866e-14,3.1663684e+16,-2.146492e-11,-223.79175,-7.298702e+30,-2.9598772e+23,-4.2341306e+33,0.010704619,-4.262323e-22,-10886.767,195.93475,2.3940731e+08,-1.3734297e-37,-2.8016076e-22,-3.492809e-11,-5039.6836,5.962697e+07,-76.06068,-8.5673265e+21,1.3207701e-23,1.562178e-37,26.532478,3.9489617e-28,1.5450501e+20,1.4179562e+37,-1.1386245e+24,2.8064455e+17,8.536562e+10,8.645561e-24,-3.4745344e-25,7.182268e-18,2.7542213e-38,-4.4767933e+11,5.8020244e+33,-1.547119e+12,-1.5278222e+32,1.3816194e-10,2.8775367e+37,-2.4745447e-19,6.779011e-27,-7.366344e-36,1.05528505e+24,4.8805834e+15,2.8588142e-06,-2.995879e+16,7.132575e+26,-2.489351e+18,4.444803e+12,8.04531e+20,27.800108,-1.1952188e+38,-3.5240968e+21,8.705135e-09,-7.4659495e+34,1.4910984e-05,-8.8790534e-29,-9.119613e+17,1.5055226e-14,-1.6081205e-13,1.4806421e-20,-4.974536e+17,-9.235154e+36,-3.9897433e-32,20.717714,-2.2660188e-29,1.229605e+18,-7.030829e-26,-0.11911836,4.7778997e-30,3.4180035e+30,-4.497392e-09,1.6916688e+11,-6.007992e-09,1810.6151,2.527266e-29,4.1154404e-24,-4.90642e+25,4.354988e-32,5.8519525e+26,-1.486255e-19,1.5037745e+29,1.6765684e-19,-3.716686e+33,121477.336,4.1158853e-31,-4.8815958e-17,4.736486e-21,6.6531276e-35,-4.115974e-23,-7.2323204e+37,3.311988e+38,7.102061e-35,0.6914403,-1.3398726e-36,15665.675,5.1442e-30,-4.0733946e+36,2.9075094e-15,9.2419216e-14,-2.0031152e+06,-4.938681e+06,1.4745451e+32,-4.2788388e-16,1.6870188e+37,51.76363,3.1371954e-09,-1.9378942e+23,503.82605,1.4615317e-19,-6.167617e+32,-2.4143207e+20,-1.4348694e+27,3.70796e+32,-6.2505974e-30,5.722064e-24,1.5823916e+14,-3.9701194e+08,7.590378e-05,37.81736,-1.2855653e-25,-2.9420562e-37,-5.01744e-35,2.0854359e+09,-2.3670766,-8.678703e-38,1.1338121e-33,-3.1916177e+33,4.800598e+17,7.39178e+08,1077.3296,2.049204e+26,5.7357483e-21,841504.2,-4.960941e-28,-3.9806363e-21,-0.0031972465,-2.721245e+33,-64867.67,-1.1708424e-36,-3.0345026e-18,0.0002853762,590381.44,-4.08793e-28,7.3273994e+15,-2.3791349e-12,-1.4723974e-22,2.9124298e+36,1.6278468e-29,-32.027534,1.6670092e+37,-1.6151843e+33,6120.9854,4.4946432e+30,5.6169963e-08,-3.2667602e-26,1.0843282e+09,1.2771826e+36,1.7270522e-25,2.4760782e+26,7.5621e+33,-1.5445818e-38,-1.4092975e-21,9.250948e+14,-6.2071345e-30,-2.4803717e-30,5.715769e+06,-1.1045412e+11,5.990224e-14,-4.9949824e+19,1.316399e-28,1.3346529e+30,-4.9172805e+35,-2.591032e-06,-2.4662888e+33,3.2015113e-38,1.8928028e+09,1.7475925e-35,-1.5167816e+14,1.3154351e+16,-1.6050726e-34,-3.81055e-24,-0.16526318,-3.771939e+30,9.118471e-36,-4.1041586e-14,-2.5451584e+21,-3.3817585e+12,-7.103881e-10,1.3891736e-36,5.5160796e+07,2.6780199e-06,1.9551739e+21,-1.855664e-33,-0.011032765,1.7574263e-16,-5.352319e+36,1.223112e-07,-5598.445,1.3935733e+10,4.606264e-18,-5.679127e+09,3.432648e+10,1.9450995e-30,1.6353132e+38,-0.018632205,8.151826e+08,0.0006464769,1.4470441e-36,3.323904e+08,0.028238963,8.3648017e+12,1.0974464e+21,7.794833e+07,5.3322546e+37,-1.3152303e-20,2.2209905e-22,-6.957394e-16,1.680201e-32,-4.5013283e-21,-5.37868e+22,3.1659847e-10,-1.5935903e+20,-2.0239142e-31,-4.397735e+15,4.042404e-18,8.289977e+22,-8.074065e+22,-1.0885104e+29,-1.7973733e+16,-3.3670788e-06,-0.27225348,1.336493e+24,-1.4551064e-08,-5.7274493e-34,3.0890197e-14,-4.083064e-14,3.0916472e-16,1.5033814e-07,-1.1776869e-37,6.540053e+34,-0.057955217,2.0857294e-28,6.9172465e+09,4.1928885e-07,3.5014566e+16,-1.26038e+18,-1.0808452,-1.7276122e+14,-11024.736,1.0925681e-15,1.6627516e-28,-1.8676024e+37,-2.0805535e+28,1.075917e-39,5.0085064e+25,-2.5898523e+11,-8.213187e-20,1.9065479e+12,-2.3956722e+14,-6.2386006e-11,-1.8336904e-20,3.4623481e-13,-7.83771e-32,32735.941,1.7740266e+25,-6.8910207e+19,-2.5651092e+09,-0.41430327,-5.7048325e-25,-9.611997e+24,0.086712606,3.580786e+32,2.7258875e-07,1.2622961e+32,1.7745434e+16,5.7896014e+34,4.6622285e+21,3.577129e-18,-1.4523347e-28,3.556452e-35,-2.6126414e-33,1.7473332e+24,4.15653e-28,9.980774e-23,-1.403635e-19,-2.0166543e-29,8.742352e-37,-9.659533e+27,1.7833492e+30,1.3324508e-18,0.00055926316,-2.0435798e-31,8.290142e+28,4.517958e+15,1.3480011e+33,-2.1913667e-22,-2.065931e-12,5.191369e+15,1.835621e-17,-7.977639e-33,-6.5399205e+20,0.19964494,-0.017151097,-0.0008125518,-8.490491e-07,1.7511816e+33,1.1756728e-18,-1.82262e+35,-1.5365792e+26,-2.4902432e-20,0.0009859746,-7.978384,2.238477e-35,-2.2113528e-16,1.7649198e+19,1.3248924e+18,-4.8978554e-11,1.3211681e+30,9.681097e+16,-7.156421e+22,2.907874e-20]':::VECTOR
						)
				)
					AS "%ptab_64" (col_129)
		)
SELECT
	count(*) AS col_130
FROM
	defaultdb.public.seed@seed__int8__float8__date_idx AS tab_65
WHERE
	tab_65._bool
GROUP BY
	tab_65._jsonb;

STMT: 31
BEGIN TRANSACTION;

STMT: 32
DELETE FROM defaultdb.public.seed AS "ta̧ b_66";

STMT: 33
INSERT
INTO
	defaultdb.public.seed AS "tab%v_67"
		(_int8, _float4, _timestamp, _timestamptz, _bool, _decimal, _string, _uuid, _jsonb)
VALUES
	(
		3493005825446431449:::INT8,
		(-0.5241608023643494):::FLOAT8,
		'2007-04-13 23:36:56.000287':::TIMESTAMP,
		NULL,
		false,
		(-86282828.765708545):::DECIMAL,
		e'XCg6\x10':::STRING,
		'7067025b-0e92-4d3a-91b4-181d01a3862f':::UUID,
		'1':::JSONB
	);

    setup_test.go:116: RESTORE
        TABLE
        	defaultdb.public.seed
        FROM
        	'http://127.0.0.1:36601/bAckup_1'
        AS OF SYSTEM TIME
        	'37 years 5 mons 881 days 15:24:40.69191':::INTERVAL
        WITH
        	into_db = 'into_db': at or near "as": syntax error: The ``RESTORE <targets> FROM <backupURI>`` syntax is no longer supported. Please use ``RESTORE <targets> FROM <subdirectory> IN <collectionURI>``.
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f82616d5c9a8c3209014f997587365ae/logTestGenerateParse4104635987
--- FAIL: TestGenerateParse (1.28s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

internal/sqlsmith.TestGenerateParse [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17595708?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17595708?buildTab=artifacts#/) on master @ [aae0a58836281ac1a57ca7fbfce6e669fadf2c45](https://github.com/cockroachdb/cockroach/commits/aae0a58836281ac1a57ca7fbfce6e669fadf2c45):


```
ALTER TABLE defaultdb.public.table_1 RENAME COLUMN col1_8 TO col_46;

STMT: 17
INSERT
INTO
	defaultdb.public.table_1 AS "t%q😲ab_32"
		(co̗l͓1_0, col1_1, " col1_2", "co l1_3", col1_4, "Col1_5", col1_6, col1_8, col1_9)
WITH
	with_10 (col_47) AS (SELECT * FROM (VALUES (NULL), (NULL)) AS tab_33 (col_47)),
	with_11 ("co�l_48")
		AS (
			SELECT
				*
			FROM
				(
					VALUES
						((-32324):::INT8),
						(15475:::INT8),
						(8591:::INT8),
						(7058:::INT8),
						(19449:::INT8),
						((-26047):::INT8)
				)
					AS tab_34 ("co�l_48")
		)
SELECT
	"t%d4ab_35".co̗l͓1_0 AS col_49,
	"t%d4ab_35".col1_4 AS col_50,
	(-497298291):::INT8 AS col_51,
	NULL AS "co%vl_52",
	"t%d4ab_35".col1_1 AS "col&_53",
	NULL AS "c%qOl_54",
	"t%d4ab_35".col1_6 AS col_55,
	"t%d4ab_35".col1_8 AS "co'l_56",
	"t%d4ab_35".col1_9 AS "co(l_57"
FROM
	defaultdb.public.table_1@"table_1_ col1_2_co l1_3_idx" AS "t%d4ab_35"
ORDER BY
	"t%d4ab_35".col1_8 NULLS LAST;

STMT: 18
COMMIT TRANSACTION;

STMT: 19
BEGIN TRANSACTION;

    setup_test.go:116: RESTORE TABLE defaultdb.public.table_1 FROM 'http://127.0.0.1:42891/bAckup_1' WITH into_db = 'into_db': at or near "with": syntax error: The ``RESTORE <targets> FROM <backupURI>`` syntax is no longer supported. Please use ``RESTORE <targets> FROM <subdirectory> IN <collectionURI>``.
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f82616d5c9a8c3209014f997587365ae/logTestGenerateParse3480007612
--- FAIL: TestGenerateParse (1.13s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

internal/sqlsmith.TestGenerateParse [failed](https://tanzanite.cluster.engflow.com/invocations/default/f6e2c6cc-8c2c-498c-b94f-01484c1eb045?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvaW50ZXJuYWwvc3Fsc21pdGg6c3Fsc21pdGhfdGVzdA==) on master @ [015b2f48cf80a6d8b60d7038c8c3457d934c716a](https://github.com/cockroachdb/cockroach/commits/015b2f48cf80a6d8b60d7038c8c3457d934c716a):


```
	JOIN defaultdb.public.table_1 AS "t ab_140" ON ("%v�tab_139"."col�1_3") = ("t ab_140"."col�1_3")
WHERE
	true
LIMIT
	43:::INT8;

STMT: 70
UPDATE
	defaultdb.public.table_1 AS "ta%vb%p_141"
SET
	col1_6 = tab_142.col1_6
FROM
	defaultdb.public.table_1 AS tab_142
WHERE
	NULL
ORDER BY
	"ta%vb%p_141".col1_7 ASC NULLS FIRST
LIMIT
	30:::INT8;

STMT: 71
DELETE FROM
	defaultdb.public.table_1 AS "	ta̷😽B_143"
USING
	defaultdb.public.table_1 AS tab_144
WHERE
	(true AND ((e'\x1a':::STRING::STRING || e'"0hbNU)\x1ct':::STRING::STRING)::STRING !~ NULL))
ORDER BY
	"	ta̷😽B_143".col1_2 ASC NULLS FIRST, "	ta̷😽B_143".crdb_internal_mvcc_timestamp
LIMIT
	7:::INT8;

STMT: 72
INSERT
INTO
	defaultdb.public.table_1 AS tab_145 (col1_0, col1_1, col1_2, "col�1_3", "col1%p_4", col1_7)
VALUES
	(
		'1981-09-29':::DATE,
		B'1000011110001',
		(-855262805280488992.5):::DECIMAL,
		873517411:::REGROLE,
		(-8679906864152564785):::INT8,
		3803372332:::REGROLE
	);

    setup_test.go:116: RESTORE TABLE defaultdb.public.table_1 FROM e'http://127.0.0.1:35421/ba\x0bc\U0001F615kup_1' WITH into_db = 'into_db': at or near "with": syntax error: The ``RESTORE <targets> FROM <backupURI>`` syntax is no longer supported. Please use ``RESTORE <targets> FROM <subdirectory> IN <collectionURI>``.
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestGenerateParse2144267220
--- FAIL: TestGenerateParse (1.34s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/internal/sqlsmith/bulkio.go
+++ b/pkg/internal/sqlsmith/bulkio.go
@@ -149,6 +149,7 @@ func makeRestore(s *Smither) (tree.Statement, bool) {
 
 	return &tree.Restore{
 		Targets: targets,
+		Subdir:  tree.NewStrVal("LATEST"),
 		From:    []tree.StringOrPlaceholderOptList{{tree.NewStrVal(name)}},
 		AsOf:    makeAsOf(s),
 		Options: tree.RestoreOptions{

